parallel.go

```diff
@@ -283,7 +290,40 @@ func (r Parallel) SearchValue(ctx context.Context, key string, opts ...ropts.Opt
 	resCh, err := r.forKey(key).search(ctx, func(ri routing.IpfsRouting) (<-chan []byte, error) {
 		return ri.SearchValue(ctx, key, opts...)
 	})
-	return resCh, err
+
+	valid := make(chan []byte)
+	var best []byte
+	go func() {
+		defer close(valid)
+
+		for v := range resCh {
+			if best == nil {
+				if r.Validator.Validate(key, v) != nil {
```

Comment:
Technically, we should be able to rely on the underlying router. Given that we don't validate anywhere else, I'm thinking we should drop this (for consistency). Thoughts?